src/celeritas/io/EventReader.hh

```diff
@@ -46,13 +46,17 @@ class EventReader
 
   public:
     // Construct from a filename
+    EventReader(std::string const& filename);
+
+    // Construct from a filename and particle params
     EventReader(std::string const& filename, SPConstParticles params);
 
     //! Prevent copying and moving
     CELER_DELETE_COPY_MOVE(EventReader);
 
     // Read a single event from the event record
     result_type operator()();
+    // Use a template and return T as HepMC3RootEvent or celeritas Primary?
```

Comment:
This file should be reverted: event reader should always require particle params.